lib/ansible/module_utils/basic.py

```diff
@@ -1649,7 +1660,7 @@ def sha256(self, filename):
         """ Return SHA-256 hex digest of local file using digest_from_file(). """
         return self.digest_from_file(filename, 'sha256')
 
-    def backup_local(self, fn):
+    def backup_local(self, fn: str | os.PathLike[str]) -> str:
```

Comment:
Should this accept other filename variants? There's no runtime checks for non-bytes, after all.
```suggestion
    def backup_local(self, fn: bytes | str | os.PathLike[bytes] | os.PathLike[str]) -> str:
```

Reply:
There isn't, but the filename of the backup will be totally screwed up if you pass `bytes` here:
```py
backupdest = '%s.%s.%s' % (fn, os.getpid(), ext)
```
You'd a file named `b'original_filename'.123.01-02-2023@01:02:03~`.

Reply:
fstrings, then `!r` should fix it either way

Reply:
That only makes the situation worse for str-based paths. If you want to fix it you need to use `to_text`/`to_native`, or use `to_bytes` and use bytes instead of str for `backupdest`.